### DISCUSSION TITLE
How to set new input values as default and reset dirtyFileds and isDirty?

### DISCUSSION THREAD

**Discussion by iBreadInside** [react-hook-form/react-hook-form#8124] [Q&A] [answered]
Hello!
I am trying to make a profile form. Default user properties are got from redux store. After changing some fields and submitting mui/TextField render helper text.

https://codesandbox.io/s/smoosh-fog-kiefqo?file=/src/form/form.jsx

What I want: 
1) after few seconds or onChange helper text hides;
2) after successful sumbit submit-button becomes disabled;

Problems: there are dependencies on isSubmitSuccessful, dirtyFields, isDirty. Most of them aren't reset after submit.
Can you please help me with that or give some advices?

I tried to use reset() but there is a problem with defaultValues. They are not rewrite to the new ones after submitting.

**Comment by bluebill1049** [comment, marked as answer]
haven you tried `reset(getValues())`?

**Reply by iBreadInside** [reply]
Thank you and sorry for my question. I am newbee.
Made some improvements for my profile form to delete password values after sumbit:
`const values = getValues();
 delete values[UserData.OLD_PASSWORD];
 delete values[UserData.NEW_PASSWORD];
 delete values[UserData.CONFIRM_PASSWORD];    
 reset(values);`
